contents/stats_vis.md

```diff
@@ -1,7 +1,181 @@
+    ax = Axis(fig[1, 1]; xticks = (1:4, levels(df.name)))
+    boxplot!(ax, levelcode.(df.name), df.grade; range=2.0, whiskerwidth=0.5)
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+Box plots can also flag anything outside the whiskers as outliers.
+By default, these observations are not shown in `Makie.jl` but you can control it with the `show_outliers` argument:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "boxplot_outliers" # hide
+    caption = "Box Plot with Outliers" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    fig = Figure(; resolution=(600, 400))
+    ax = Axis(fig[1, 1]; xticks = (1:4, levels(df.name)))
+    boxplot!(ax, levelcode.(df.name), df.grade; range=0.5, show_outliers=true)
+    Options(current_figure(); filename=label, caption, label) # hide
+    """
+sco(s)
+```
+
+As you can see, **box plots are a useful way to visualize data with robust central tendencies and dispersion measures to outliers**.
+
 ### Density Plots {#sec:stats_vis_densityplots}
 
+Box plot limit us just to summary statistics like median, quartiles and IQRs.
+Often we want to see the full distribution of our data.
+This can be accomplished with **density plots**.
+**Density plots are graphical density estimations of numerical data**.
+It shows us the full distribution of a given variable by depicting it as a density, where the higher the curve at a given point more likely is the variable to take certain value.
+
+The density plot can also be drawn using `Makie.jl`, however it is more convoluted than the box plot.
+First, we want to pass for each `density!` function only the values with respect to one observation.
+Thus, we define a `values` function that will accept a `code` argument to filter the dataset's variable `name` wrapped with the `levelcode` function.
+Then, we plot a density `pltobj` for each one of the variable `name`'s `levels`.
+Finally, we make sure that the density `plotobj`s have their own `ytick` with the `offset` keyword paired with a custom `yticks` in the `Axis` constructor by specifying, same as before, a tuple of values and labels.
+The effect of the `offset` in the `for` loop is the increment from 1 to 4, by 1, of both the `offset` argument for `density!` and the `code` argument for `values`:
+
+```jl
+s = """
+    CairoMakie.activate!() # hide
+    label = "densityplot" # hide
+    caption = "Density Plot" # hide
+    df = more_grades()
+    transform!(df, :name => categorical; renamecols=false)
+    categories = levels(df.name)
+    values(code) = filter(row -> levelcode.(row.name) == code, df).grade
+    fig = Figure(; resolution=(600, 400))
+    ax = Axis(fig[1, 1]; yticks = (1:4, categories), limits=((-1, 11), nothing))
+    for i in 1:length(categories)
+        density!(ax, values(i); offset=i)
+    end
+    Options(current_figure(); filename=label, caption, label) # hide
```

Comment:
Maybe an annoying comment, but can something be done about the top (Sally) in both these figures? It looks very weird:

![image](https://user-images.githubusercontent.com/20724914/134229502-c3ce10d4-7e80-40eb-a6b6-1f3cca576c67.png)

Reply:
@lazarusA what you think?

Reply:
sure, maybe some `scale_to =` will fix it, or a lower offset. Tomorrow I will check... today no more code.

Reply:
it seems that `density` does not have a `scale_to = ` arg, so my suggestion will be to have a different offset. But still that distribution is Sally's doing. Which is also ok.

Reply:
Poor Sally, @rikhuijzer gave her a grade 1 (from 0 to 10)